packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md

```diff
@@ -1,154 +1,144 @@
 Launch Xcode and create a new iOS app project using SwiftUI for the interface and Swift for the language.
 
-> **Note:** The sample code is compatible with iOS 14 and later.
+> **Note:** The sample code uses features of SwiftUI available in iOS 14, and the async/await feature of iOS 15. Use the completion handler version of the async/await functions to add compatibility with iOS 14 and later.
 
 #### Add the UI
 
-The main view implements all of the app functionality, including initializing the Okta SDK. Note that best practice for a full app uses a singleton to manage the interaction with the SDK.
+The functionality of the app is implemented in the main SwiftUI view, including view state, view model updates, and data model. For a production app, consider partitioning the view and data models into singletons, and then using Combine, bindings, or `@Environment` for sharing state.
 
-Text views display the current state and any requested information, such as the access token. The available buttons depend on the sign-in state of the user. Update the imported modules and replace the `ContentView` `struct` with the following code:
+The text views in the app display its current state and any optional information, such as the access token. The available buttons depend on the sign-in state of the user. The top-level `ZStack` contains a full-screen view that prevents the user interacting with the UI while the app is performing certain asynchronous operations, such as reading the user information. A production app may also perform asynchronous operations that do require showing a busy indicator but don't require blocking the UI.
+
+The following image shows the completed app in both the sign-in and signed-on states:
+
+<div class="full border">
+
+![Two iPhone screenshots that show the signed-out and signed-in UI.](/img/sign-users-in/redirect-authentication/ios/app-ui-ios.png)
+
+</div>
+
+<!--
+Source image(s): https://www.figma.com/file/i3huE0gEoISu2evquOq5yJ/app-ui-redirect-ios?node-id=0%3A1
+There's only one board. The group for the image is: "Side-by-side screenshots"
+-->
+
+Open `ContentView.swift` and replace it with the following code:
 
 ```swift
 import SwiftUI
-import UIKit
+import AuthFoundation
+import WebAuthenticationUI
 
 struct ContentView: View {
-  @State var showingError: Bool = false
-  @State var errorMessage: String? = nil
-  @State var errorTitle: String? = nil
-  @State var statusText:String?  = "Not signed in."
-  @State var infoText: String? = ""
-  @State var signedIn: Bool = false
-
-  var body: some View {
-    VStack() {
-      HStack {
-        if !signedIn {
-          Button("Sign In") {
-            signIn()
-          }
-          .padding()
-        } else {
-          VStack {
-            Button("Sign Out") {
-              signOut()
-            }
-            .padding(.bottom)
+   // Used for displaying error messages.
```

Comment:
Refresh my memory re: comments in code - if it's a fragment, do we add a period?
```suggestion
   // Used for displaying error messages.
```

Reply:
I seem to remember from a previous article reviewed by Susan that the answer is "yes" and I can't find it in any of the various and sundry Okta style guides. I'm going to leave this for now. If there's a style issue, Susan will catch it in that final pass.